dice-where-downloader-lib/src/main/java/technology/dice/dicewhere/downloader/destination/local/LocalFileAcceptor.java

```diff
@@ -72,9 +71,6 @@ public Optional<MD5Checksum> existingFileMd5() {
       try (InputStream is = Files.newInputStream(this.destination);
           BufferedInputStream bis = new BufferedInputStream(is);
           StreamWithMD5Decorator md5Is = StreamWithMD5Decorator.of(bis)) {
-        byte[] buffer = new byte[BUFFER];
-        while ((md5Is.read(buffer)) != -1) {
-        }
```

Comment:
@julianhowarth the digest is incrementally added per each byte we read. 
To be able to generate and compare the hash we need to consume every byte for the stream and then perform the MD5. Because we cannot re-use the same InputStream, what we're doing here is copying the stream ( therefore updating the digest for the original stream ) and provide a copied InputStream to be streamed elsewhere - that contains the exact md5 checksum of the original one.